project-base/build.xml

```diff
@@ -52,10 +52,10 @@
     </if>
 
     <target name="build" depends="build-deploy-part-1-db-independent, build-deploy-part-2-db-dependent" description="Builds application for production preserving your DB."/>
-    <target name="build-deploy-part-1-db-independent" depends="clean,composer,npm,dirs-create,assets" description="First part of application build for production preserving your DB (can be run without maintenance page)."/>
+    <target name="build-deploy-part-1-db-independent" depends="clean,clean-redis,composer,npm,dirs-create,assets" description="First part of application build for production preserving your DB (can be run without maintenance page)."/>
```

Comment:
This can be an issue if you will be deploying new release and someone will approach to the portal. At that moment he writes to redis cache informations that may not be consistent with new release. After deploy application can throws error.

Reply:
We preceed this by creating a prefix that is generated during a build. It's working like generating css files. So we have file with that prefix and with phing target, we change that version
```
parameters:
    build-version: %%version%%
```

```
<target name="generate-build-version">
    <exec executable="${path.php.executable}" checkreturn="true" outputProperty="version">
        <arg value="-r" />
        <arg value="echo uniqid(true);" />
    </exec>
    <copy file="${path.app}/config/parameters_version.yml.dist" tofile="${path.app}/config/parameters_version.yml" overwrite="true">
        <filterchain>
            <replacetokens begintoken="%%" endtoken="%%">
                <token key="version" value="${version}" />
            </replacetokens>
        </filterchain>
    </copy>
</target>
```

Reply:
I agree with this because we don't want any cache errors after deployment

Reply:
Now I don't agree with this and I think it should be part of the PR or issue as I mentioned above